Bullet Generation and Behaviour & Shoot use case

rachphm opened the pull request:
Implemented shooting use case triggered by pressing T.

Updated entityFactory.create() to return an entity ID (to enable the use of loadEntity(id))

TODO:

-  Adjust the bullet spawn position to originate from the bottom-right corner of the camera.

- Make sure bullet meshes are properly discarded upon collision or after a certain time.

Emanon-tt committed:
use viewmanager to control the switch between login and register

Emanon-tt committed:
switch failed

rachphm committed:
Added ShootInteractor

rachphm committed:
Added ShootInteractor

rachphm committed:
Added ShootInteractor

zaynaha reviewed (commented):
it does not run for me
<img width="1363" height="324" alt="Screenshot 2025-11-30 at 2 34 12 PM" src="https://github.com/user-attachments/assets/c136383b-b74c-46ac-80aa-177ef5f7871e" />
does it work on your end

rachphm committed:
Added ShootInteractor

rachphm commented:
Mb I forgot to add the model file, can you update and try again?

Emanon-tt committed:
still some issue with the gameview

KurbyDoo reviewed (approved):
looks good, i have some regrets about how the world sync was handled but i think its too late to go back and fix them, so this is ok for now

KurbyDoo commented on core/src/main/java/application/game_use_cases/shoot/ShootInteractor.java:
```diff
@@ -0,0 +1,24 @@
+    }
+
+    @Override
+    public int execute(ShootInputData inputData) {
```
i dont really like how the id is passed here instead of an ouput data object, but i guess its fine for now

KurbyDoo reviewed (changes requested):
Actually, sorry can you rebase and add a test case. Thanks!

Emanon-tt committed:
I changed the UI from Swing to the LibGDX and that's work

rachphm committed:
Added ShootOutputData

KurbyDoo committed:
removed arch violations from GameSimulationController and WorldSyncController

rachphm committed:
Added ShootOutputData

rachphm committed:
Added ShootOutputData

rachphm committed:
Added BulletBehaviourTest and ShootInteractorTest

rachphm committed:
Added BulletBehaviourTest and ShootInteractorTest

rachphm committed:
Update GameView

rachphm commented:
BulletBehaviourTest
<img width="620" height="572" alt="image" src="https://github.com/user-attachments/assets/3d751dc5-d0a5-446d-b90e-eaec7d1adaac" />

KurbyDoo reviewed (approved):
thanks!

Leobstere reviewed (commented):
Bullet might want to take an damage parameter from the weapon being used.

rachphm commented:
Modified InputData to contain a boolean value indicating if the player is allowed to shoot.
Updated bullet behaviour so it travels in the direction that the player is looking.
